ruby/spec/find_cache_spec.rb

```diff
@@ -48,4 +48,20 @@
     expect(result).to be_success
     expect(result.resolved_key).to be_nil
   end
+
+  it 'respects key order' do
+    keys = [
+      'v1-bundler-cache-linux-x86_64-branch-name-',
+      'v1-bundler-cache-linux-x86_64-'
+    ]
+
+    result = FindCache.call(
+      keys: keys,
+      bucket: bucket,
+      prefix: prefix
+    )
+
+    expect(result).to be_success
+    expect(result.resolved_key).to eq 'v1-bundler-cache-linux-x86_64-branch-name-deadbeef'
```

Comment:
Just to make sure that I understand this correctly if we did not respect the key order, this would have instead matched:
```
v1-bundler-cache-linux-x86_64-deadbeef
```

Reply:
No it would have matched either `longer-branch-name` or one of the really long keys from the checksum tests.

(In this case it's the latter but I didn't want to couple the data from those tests together in case I change something later.)